shiny/session/_session.py

```diff
@@ -390,7 +395,7 @@ def _is_hidden(self, name: str) -> bool:
 
     async def _dispatch(self, message: ClientMessageOther) -> None:
         try:
-            func = self._message_handlers[message["method"]]
+            async_func = self._message_handlers[message["method"]]
```

Comment:
Store both the session and the function here, so that we can use `with_session_context()` below.